core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -443,6 +443,21 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(list2DiskGet.get.readMethod === DataReadMethod.Disk)
   }
 
+  test("block manager crash test") {
+    conf.set("spark.network.timeout", "5s")
+    store = makeBlockManager(8000)
+    store2 = makeBlockManager(8000)
+    val list1 = List(new Array[Byte](4000))
+    store2.putIterator("list1", list1.iterator, StorageLevel.MEMORY_ONLY, tellMaster = true)
+    val list1Get = store.getRemote("list1")
+    assert(list1Get.isDefined, "list1get expected to be fetched")
+    // simulate block manager crashed
+    store2.stop()
+    val list1GetAgain = store.getRemoteBytes("list1")
+    assert(!list1GetAgain.isDefined, "list1getagain expected to be not fetched")
+    conf.set("spark.network.timeout", "120s")
```

Comment:
This statement looks like it should be unnecessary but I guess it's not since it looks like this test is mutating a shared configuration. Is there any way to avoid this type of coupling between tests or to ensure that the previous state is restored in a `finally` block or `afterEach()` method?